### DISCUSSION TITLE
Is there a way to set max connections on the SQS client?

### DISCUSSION THREAD

**Discussion by jadlr** [aws/aws-sdk-kotlin#1056] [Q&A] [answered]
Java has an options that is explained here: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-throughput-horizontal-scaling-and-batching.html

```java
final AmazonSQS sqsClient = AmazonSQSClientBuilder.standard()
        .withClientConfiguration(new ClientConfiguration()
                .withMaxConnections(producerCount + consumerCount))
        .build();
```

I could not find this option in the kotlin version.

**Comment by ianbotsf** [comment, marked as answer]
Hi @jadlr. The AWS SDK for Kotlin does not support explicit configuration of maximum _connections_ due to limitations in the underlying default HTTP engine (OkHttp). That said, it is possible to configure _concurrency_ during client initialization:

```kotlin
val sqs = SqsClient {
    httpClient {
        maxConcurrency = 64u
    }
}
```

Does that address your needs?

**Reply by jadlr** [reply]
Yes, absolutely! Thanks for your reply @ianbotsf
